server/src/main/java/org/elasticsearch/index/analysis/ShingleTokenFilterFactory.java

```diff
@@ -63,6 +63,10 @@ public TokenStream create(TokenStream tokenStream) {
         return factory.create(tokenStream);
     }
 
+    @Override
+    public TokenFilterFactory getSynonymFilter(boolean lenient) {
+        return IDENTITY_FILTER;
```

Comment:
this will prevent all rules to match ?

Reply:
Right, it should return `this` if `lenient=true`, or if it produces shingles of a fixed size and no unigrams; otherwise it should throw an exception, or log a deprecation warning and return `this`.

Reply:
Again, writing tests, this will work if the inputs and outputs of the synonym map have the same lengths, but will otherwise break.  For example, if we map `great big` to `massive`, and shingle things correctly, then the string `a great big house`, put through a shingle filter and a synonym filter, will produce the output `a great`, `great big`, `massive`, `big house`; searches for `massive house` will fail.  The shingle filter, like the common grams filter, should only really be placed at the end of a filter chain.  So I think we should throw an error unless `lenient=true`

Reply:
+1